docs/developer/update-v1alpha1-provider-to-v1alpha2-infra-provider.md

```diff
@@ -0,0 +1,188 @@
+}
+```
+
+If present, this field is automatically inspected in order to obtain an endpoint at which the Kubernetes cluster may be accessed.
+
+## Create the infrastructure controllers
+
+The actuator model from v1alpha1 has been replaced by the infrastructure controllers in v1alpha2:
+
+| v1alpha1 | v1alpha2 |
+|---|---|
+| [`sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/cluster.Actuator`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/e6a57dc61826b8c7806eba22a513c9722c420754/pkg/cloud/aws/actuators/cluster/actuator.go#L50-L57) | [`sigs.k8s.io/cluster-api-provider-aws/pkg/controller/awscluster.ReconcileAWSCluster`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L98-L103) |
+| [`sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine.Actuator`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/e6a57dc61826b8c7806eba22a513c9722c420754/pkg/cloud/aws/actuators/machine/actuator.go#L57-L65) | [`sigs.k8s.io/cluster-api-provider-aws/pkg/controller/awsmachine.ReconcileAWSMachine`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awsmachine/awsmachine_controller.go#L104-L109) |
+
+### The provider cluster controller
+
+Instead of processing the CAPI `Cluster` resources like the old actuator model, the new provider cluster controller operates on the new provider `Cluster` CRD. However, the overall workflow should feel the same as the old cluster actuator. For example, take the AWS cluster controller's [reconcile function](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L105-L162), it:
+
+1. Fetches the [AWS cluster resource](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L113-L121):
+
+    ```golang
+    awsCluster := &infrastructurev1alpha2.AWSCluster{}
+    err := r.Get(ctx, request.NamespacedName, awsCluster)
+    if err != nil {
+        if apierrors.IsNotFound(err) {
+            return reconcile.Result{}, nil
+        }
+        return reconcile.Result{}, err
+    }
+    ```
+
+2. [Fetches the CAPI cluster resource](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0a05127734a4fb955742b27c6e326a65821851ce/pkg/controller/awscluster/awscluster_controller.go#L125-L133) that has a one-to-one relationship with the AWS cluster resource:
+
+    ```golang
+    cluster, err := util.GetOwnerCluster(ctx, r.Client, awsCluster.ObjectMeta)
+    if err != nil {
+        return reconcile.Result{}, err
+    }
+    if cluster == nil {
+        logger.Info("Waiting for Cluster Controller to set OwnerRef on AWSCluster")
+        return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
We shouldn't need to requeue - we need the AWSCluster to get updated with an ownerRef to the Cluster, which means we'll process it again as part of a regular reconcile event (https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1037)